lib/checkunusedvar.cpp

```diff
@@ -1586,3 +1589,67 @@ bool CheckUnusedVar::isEmptyType(const Type* type)
     emptyType=false;   // unknown types are assumed to be nonempty
     return emptyType;
 }
+
+bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func) {
```

Comment:
it seems to me `isFunctionWithoutSideEffects` can be a static function.

Reply:
It uses non-static `CheckUnusedVar::isVariableWithoutSideEffects()`, which uses also non-static `CheckUnusedVar::isRecordTypeWithoutSideEffects()`. Which also may use some class member functions. 
Do you want me to try make all of them static?